components/search/src/filter.rs

```diff
@@ -191,7 +207,10 @@ impl Filter for Vec<JSONSearchConfigurationRecords> {
         for record in self {
             match record {
                 JSONSearchConfigurationRecords::Engine(engine) => {
-                    let result = maybe_extract_engine_config(user_environment, engine.clone());
+                    let mut result = maybe_extract_engine_config(user_environment, engine.clone());
+                    if let Some(overrides_data) = &overrides {
+                        apply_overrides(&mut result, overrides_data);
+                    }
```

Comment:
Currently, I think this is being called every time we add a new engine to the result - each time around the loop.

I think it should be called after the loop so that we only loop over the engines once for each overrides record.